packages/bpk-component-snippet/src/BpkSnippet.tsx

```diff
@@ -0,0 +1,165 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { MouseEvent } from 'react';
+
+import { BpkButtonV2, BUTTON_TYPES } from '../../bpk-component-button';
+import BpkImage, {
+  BORDER_RADIUS_STYLES,
+  withLazyLoading,
+} from '../../bpk-component-image';
+import BpkText, {
+  TEXT_STYLES,
+  type TextStyle,
+} from '../../bpk-component-text/src/BpkText';
+import { cssModules } from '../../bpk-react-utils';
+
+import type { ButtonType } from '../../bpk-component-button/src/BpkButtonV2/common-types';
+
+import STYLES from './BpkSnippet.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export const BODY_STYLE = {
+  bodyDefault: TEXT_STYLES.bodyDefault,
+  bodyLongform: TEXT_STYLES.bodyLongform,
+} as const;
+
+export const DESKTOP_LAYOUT = {
+  imageLeft: 'imageLeft',
+  imageRight: 'imageRight',
+  vertical: 'vertical',
+} as const;
+
+export const IMAGE_ORIENTATION = {
+  landscape: 'landscape',
+  portrait: 'portrait',
+  square: 'square',
+} as const;
+
+export type Props = {
+  src: string;
+  altText: string;
+  headline?: string | null;
+  subheading?: string | null;
+  bodyText?: string | null;
+  buttonText?: string | null;
+  bodyStyle?: (typeof BODY_STYLE)[keyof typeof BODY_STYLE];
+  buttonStyle?: ButtonType;
+  headlineStyle?: TextStyle;
+  desktopLayout?: (typeof DESKTOP_LAYOUT)[keyof typeof DESKTOP_LAYOUT];
+  imageOrientation?: (typeof IMAGE_ORIENTATION)[keyof typeof IMAGE_ORIENTATION];
+  imageRadius?: boolean;
+  onClick?: (event: MouseEvent<HTMLButtonElement & HTMLAnchorElement>) => void;
+};
+
+const documentIfExists = typeof window !== 'undefined' ? document : null;
+const LazyLoadedImage = withLazyLoading(BpkImage, documentIfExists);
+
+const BpkSnippet = ({
+  altText,
+  bodyStyle = TEXT_STYLES.bodyDefault,
+  bodyText,
+  buttonStyle = BUTTON_TYPES.primary,
+  buttonText,
+  desktopLayout = 'imageLeft',
+  headline,
+  headlineStyle = TEXT_STYLES.hero5,
+  imageOrientation,
+  imageRadius = true,
+  onClick,
+  src,
+  subheading,
+}: Props) => {
+  let imageAspectRatio;
+  switch (imageOrientation) {
+    case 'landscape':
+      imageAspectRatio = 3 / 2;
+      break;
+    case 'square':
+      imageAspectRatio = 1;
+      break;
+    case 'portrait':
+      imageAspectRatio = 5 / 7;
+      break;
+    default:
+      imageAspectRatio = 3 / 2;
+      break;
+  }
+  return (
+    <div
+      className={getClassName(
+        'bpk-snippet',
+        desktopLayout === 'imageRight' && 'bpk-snippet--row-reverse',
+        desktopLayout === 'vertical' && 'bpk-snippet--vertical',
+      )}
+    >
+      <div
+        className={getClassName(
+          desktopLayout === 'vertical'
+            ? 'bpk-snippet--vertical--container'
+            : 'bpk-snippet--container',
+        )}
+      >
+        <LazyLoadedImage
+          // eslint-disable-next-line @skyscanner/rules/forbid-component-props
```

Comment:
I know passing `className` is against the rules, but in BpkImage, the img is wrapped by a div with `height: 0px; padding-bottom: 100%;` style, so when the image height is shorter than padding-bottom, there will be a big blank at the bottom.
I see many in business code they have to cover img styles like this, and it's really bad, but there's no other way. Unless I go back to native img
<img width="802" alt="image" src="https://github.com/user-attachments/assets/dd3eaed9-c620-405e-9380-f493e1ea10ee">